Mathlib/Data/Polynomial/Derivation.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2023 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Richard Hill
+-/
+import Mathlib.Data.Polynomial.Derivative
+import Mathlib.RingTheory.Derivation.Basic
+import Mathlib.Data.Polynomial.AlgebraMap
+
+/-!
+# Derivations of univariate polynomials
+
+In this file we prove that an `R`-derivation of `Polynomial R` is determined by its value on `X`.
+We also provide a constructor `Polynomial.mkDerivation` that
+builds a derivation from its value on `X`, and a linear equivalence
+`Polynomial.mkDerivationEquiv` between `A` and `Derivation (Polynomial R) A`.
+-/
+
+namespace Polynomial
+
+open scoped BigOperators
+
+noncomputable section
+
+variable {R A : Type _} [CommSemiring R] [AddCommMonoid A] [Module R A] [Module (Polynomial R) A]
+
+section
+
+@[simp]
+theorem derivation_C (D : Derivation R (Polynomial R) A) (a : R) : D (C a) = 0 :=
```

Comment:
How easily does this theory generalize to `Derivation R (Polynomial S) A`?

Reply:
This PR is showing that an R-derivation on R[X] is determined by its value on X, and this is true because R[X] is generated as an R-algebra by R (on which the R-derivation has to be 0 by the axioms of a derivation) and X. If you consider R-derivations on S[X] with S an R-algebra then this argument no longer works because there can be elements of S which don't derive to 0 in this generality. So for example `derivation_C` would be false for `a : S`, `derivation_ext` would be false etc.

Reply:
I also don't see a useful generalisation here.